### PR TITLE
Bump version, reset build number for jsoncpp v1.8.4

### PR DESCRIPTION
**Pull request by chambbj**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by chambbj** [comment]
@conda-forge-admin, please rerender

**Comment by jakirkham** [comment]
~~This may no longer compile with VS 2008. Would recommend skipping that build and re-rendering.~~

NVM found an upstream patch that fixes this issue.

xref: https://github.com/open-source-parsers/jsoncpp/pull/695#issuecomment-353173044

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,10 +7,17 @@ package:
...
   url: https://github.com/open-source-parsers/jsoncpp/archive/{{ version }}.tar.gz
-  sha256: 858db2faf348f89fdf1062bd3e79256772e897e7f17df73e0624edf004f2f9ac
+  sha256: c49deac9e0933bcb7044f08516861a2d560988540b23de2ac1ad443b219afdb6
+  patches:
+    ###################################################################
+    #  Include upstreamed patch for VS 2008 support.                  #
+    #                                                                 #
+    #  xref: https://github.com/open-source-parsers/jsoncpp/pull/733  #
+    ###################################################################
+    - PR_733.patch
```
Comment (jakirkham) [inline recipe/meta.yaml:17]: xref: https://github.com/open-source-parsers/jsoncpp/pull/733

**Comment by chambbj** [comment]
Thanks for locating the missing piece @jakirkham. LGTM

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render.

**Comment by jakirkham** [comment]
Strange, re-rendering seems to have added a `python` matrix even though `python` is in `requirements/build` and not in `requirements/host`, which also exists. Any ideas?

cc @isuruf

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render.

**Comment by jakirkham** [comment]
Seems adding an empty, explicit `requirements/run` section addressed the issue. Though it doesn't seem like this should have mattered.

**Comment by jakirkham** [comment]
Raised issue ( https://github.com/conda/conda-build/issues/2989 ) upstream about the `python` matrix that showed up.

**Comment by jakirkham** [comment]
Thanks @chambbj :)

**Pull request merged**